api/v1/utils/pattern.ts

```diff
@@ -0,0 +1,44 @@
+export default class Pattern {
+  private pattern: number;
+  private date: Date;
+  private day: number;
+
+  constructor() {
+    this.pattern = -1; //Initialize as an incorrect value to force new assignment
+    this.date = new Date(new Date(this.getJPTime())); //Need JP time
```

Comment:
Is there a need to wrap the JP time in `new Date(..)` twice?

Reply:
Nope! This was a relic of the function code originally being in the constructor. Fixed in 20fde0bb6b01f8564e5138260c15bdd1c4fed5de